src/test/datascience/notebook/helper.ts

```diff
@@ -763,3 +767,37 @@ export async function hijackPrompt(
         clickButton: (text?: string) => clickButton.resolve(text || buttonToClick?.text)
     };
 }
+
+export async function asPromise<T>(
+    event: Event<T>,
+    timeout = env.uiKind === UIKind.Desktop ? 5000 : 15000
+): Promise<T> {
+    return new Promise<T>((resolve, reject) => {
+        // eslint-disable-next-line @typescript-eslint/no-use-before-define
+        const handle = setTimeout(() => {
+            sub.dispose();
+            reject(new Error('asPromise TIMEOUT reached'));
+        }, timeout);
+        const sub = event((e) => {
+            clearTimeout(handle);
+            sub.dispose();
+            resolve(e);
+        });
+    });
+}
+
+export async function waitForEvent<T>(
+    eventType: string,
+    debugAdapter: KernelDebugAdapter,
+    timeout = env.uiKind === UIKind.Desktop ? 5000 : 15000
+): Promise<T> {
+    setTimeout(() => {
+        return;
```

Comment:
I don't think this timeout will do anything. It should be throwing an exception or rejecting a promise.

Reply:
I'll throw an exception